### DISCUSSION TITLE
Mounting local folder in docker will break tutorial DAG

### DISCUSSION THREAD

**Discussion by srggrs** [apache/airflow#17455] [Q&A]
Hi, 

I just installed Airflow 2.1.2 via Docker as per [official instructions](https://airflow.apache.org/docs/apache-airflow/stable/start/docker.html#). Then I run the initialisation command and `docker-compose up` as per documentation. I also save the [tutorial DAG](https://airflow.apache.org/docs/apache-airflow/stable/tutorial.html) in the `./dag` folder as `my_first_tutorial`.

My issue is that when I mount a local folder on the celery worker with this line:
```docker-compose
airflow-worker:
  ...
  volumes:
    - /path/to/my/folder:/home/airflow/work
```
The tutorial DAG fails and I can see the following error in the terminal:
```bash
airflow-worker_1     | [2021-08-06 00:02:57,884: ERROR/ForkPoolWorker-15] Failed to execute task dag_id could not be found: my_first_tutorial. Either the dag did not exist or it failed to parse..
airflow-worker_1     | Traceback (most recent call last):
airflow-worker_1     |   File "/home/airflow/.local/lib/python3.6/site-packages/airflow/executors/celery_executor.py", line 117, in _execute_in_fork
airflow-worker_1     |     args.func(args)
airflow-worker_1     |   File "/home/airflow/.local/lib/python3.6/site-packages/airflow/cli/cli_parser.py", line 48, in command
airflow-worker_1     |     return func(*args, **kwargs)
airflow-worker_1     |   File "/home/airflow/.local/lib/python3.6/site-packages/airflow/utils/cli.py", line 91, in wrapper
airflow-worker_1     |     return f(*args, **kwargs)
airflow-worker_1     |   File "/home/airflow/.local/lib/python3.6/site-packages/airflow/cli/commands/task_command.py", line 220, in task_run
airflow-worker_1     |     dag = get_dag(args.subdir, args.dag_id)
airflow-worker_1     |   File "/home/airflow/.local/lib/python3.6/site-packages/airflow/utils/cli.py", line 193, in get_dag
airflow-worker_1     |     'parse.'.format(dag_id)
airflow-worker_1     | airflow.exceptions.AirflowException: dag_id could not be found: my_first_tutorial. Either the dag did not exist or it failed to parse.
airflow-worker_1     | [2021-08-06 00:02:57,893: ERROR/ForkPoolWorker-15] Task airflow.executors.celery_executor.execute_command[80db90d2-2578-4ce4-895b-39659f7ee253] raised unexpected: AirflowException('Celery command failed on host: 2a71e2407442',)
airflow-worker_1     | Traceback (most recent call last):
airflow-worker_1     |   File "/home/airflow/.local/lib/python3.6/site-packages/celery/app/trace.py", line 412, in trace_task
airflow-worker_1     |     R = retval = fun(*args, **kwargs)
airflow-worker_1     |   File "/home/airflow/.local/lib/python3.6/site-packages/celery/app/trace.py", line 704, in __protected_call__
airflow-worker_1     |     return self.run(*args, **kwargs)
airflow-worker_1     |   File "/home/airflow/.local/lib/python3.6/site-packages/airflow/executors/celery_executor.py", line 88, in execute_command
airflow-worker_1     |     _execute_in_fork(command_to_exec)
airflow-worker_1     |   File "/home/airflow/.local/lib/python3.6/site-packages/airflow/executors/celery_executor.py", line 99, in _execute_in_fork
airflow-worker_1     |     raise AirflowException('Celery command failed on host: ' + get_hostname())
airflow-worker_1     | airflow.exceptions.AirflowException: Celery command failed on host: 2a71e2407442
```
From the UI, clicking on the task logs I get this:
```
*** Log file does not exist: /opt/airflow/logs/my_first_tutorial/print_date/2021-08-06T00:02:57.038239+00:00/1.log
*** Fetching from: http://:8793/log/my_first_tutorial/print_date/2021-08-06T00:02:57.038239+00:00/1.log
*** Failed to fetch log file from worker. The request to ':///' is missing either an 'http://'                         or 'https://' protocol.
```

Is this some weirdness of docker/docker-compose? Maybe because I overwrite the `/home/airflow` folder with my mounted, so the Python installation will be overwritten ( `/home/airflow/.local/...`)?

Would be great to have some suggestions in the documentation where to mount our folder with our scripts.

When I remove that mounting volume line, everything works fine.

Thanks for listening.
Cheers

**Comment by potiuk** [comment]
I cannot reproduce it. I mounted my own worker with adding the last line and I have no problems whatsoever :

```
  airflow-worker:
    <<: *airflow-common
    command: celery worker
    healthcheck:
      test:
        - "CMD-SHELL"
        - 'celery --app airflow.executors.celery_executor.app inspect ping -d "celery@$${HOSTNAME}"'
      interval: 10s
      timeout: 10s
      retries: 5
    restart: always
    volumes:
      - /home/jarek/:/home/airflow/work
```

Tutorial DAAGS are running fine:

![Screenshot from 2021-08-06 12-26-15](https://user-images.githubusercontent.com/595491/128497004-b60bfde0-ac18-41cf-bbd0-957fbc6b3331.png)

You must have done some other modifications in your setup. Cleanup and reinitialize everything from the scratch following the official tutorial. Note that the "docker-image" is just a "quick-start" to get you running. It will run if you follow precisely the instructions set from the clean state, but If you make any modification to it, changed perrmissions, added volumes etc. , you must be very sure to know what you are doing. There is a lot of information on where airflow expects to have some directories in the airflow documentation. 

If you do not know exactly what you are doing with modifying docker compose, I recommend you to put any files you want to share with Airflow inside the folders that are already there ("dags", "plugins").
